added updates to messaging

chenchik opened the pull request:
added random byte messaging and written message messaging.

chenchik committed:
added updates to messaging

mdaum commented on udark/udapp/src/main/java/io/underdark/app/model/Node.java:
```diff
@@ -151,12 +156,12 @@ public void transportLinkConnected(Transport transport, Link link)
 		activity.refreshPeers();
 		activity.refreshButtons();
-		String rt = encodeRoutingTable();
+		//String rt = encodeRoutingTable();
```
Why are we now not encoding and sending out the routing table when we connect with a link? Without this, no one will ever get a message w/ protocol 0

mdaum commented on udark/udapp/src/main/java/io/underdark/app/model/Node.java:
```diff
@@ -119,9 +123,10 @@ public void sendFrame(byte[] frameData, Link link){
 	public void sendFrame(Link link, int protocolId, String str){
-		String newStr = Integer.toString(protocolId) + str;
+		//protocol id | sender id | reciever id | payload
+		String newStr = Integer.toString(protocolId) + "|" + link.getNodeId() + "|" + this.nodeId + "|" + str;
```
We should only attach this metadata if we are not in protocol 0 or 1. When we exchange routing information, we will only ever communicate with people within our naturally discovered links. No need for this extra information/parsing of it on the receiving end when in modes 0 or 1.

mdaum replied:
nvm was tired when i made this comment....misread

mdaum replied:
lol wait no I'm right we only append the meta info when not in mode 0 or 1

mdaum replied:
Also, if you always put in link.getNodeId, you lose the original sender of the message after 1 hop

mdaum commented on udark/udapp/src/main/java/io/underdark/app/model/Node.java:
```diff
@@ -151,12 +156,12 @@ public void transportLinkConnected(Transport transport, Link link)
+		//String rt = encodeRoutingTable();
 		//now send encoded routing table to each adjacent node
-		for (Link l:
+		/*for (Link l:
```
Same question as previous comment. We should be sending this routing info to all links.

mdaum reviewed (changes requested):
We will make the fixes together Monday afternoon and make sure everything works with 3+ phones.
Let's hold off on merging this for now.

mdaum commented on udark/udapp/src/main/java/io/underdark/app/model/Node.java:
```diff
@@ -218,18 +233,30 @@ public void transportLinkDidReceiveFrame(Transport transport, Link sender, byte[
+			//actual message, could be random or typed
 			case 2:
-				break; //nothing for now....this is just random data
+				messagePieces = message.split("\\|");
```
So we show the message if we are meant to get it....sure....but if now we need to send to the person in our routing table mapped to the destination!

mdaum committed:
fixed bugs in pr, and segragated some of the logic to make things a bit
easier to read and expand upon later....everything in this branch is
working as is now. Message hopping works.

mdaum reviewed (approved):
I fixed up the PR. Take a look at my commit which goes on top of yours to see what I changed. It all works so lemme know if you have questions.
